fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1262,6 +1541,44 @@ TEST_CASE("noSim/fdbserver/KeyValueStoreRocksDB/RocksDBReopen") {
 	return Void();
 }
 
+TEST_CASE("noSim/fdbserver/KeyValueStoreRocksDB/CheckpointRestore") {
+	state std::string cwd = platform::getWorkingDirectory() + "/";
+	state std::string rocksDBTestDir = "rocksdb-kvstore-br-test-db";
+	platform::eraseDirectoryRecursive(rocksDBTestDir);
+
+	state IKeyValueStore* kvStore = new RocksDBKeyValueStore(rocksDBTestDir, deterministicRandom()->randomUniqueID());
+	wait(kvStore->init());
+
+	kvStore->set({ LiteralStringRef("foo"), LiteralStringRef("bar") });
+	wait(kvStore->commit(false));
+
+	Optional<Value> val = wait(kvStore->readValue(LiteralStringRef("foo")));
+	ASSERT(Optional<Value>(LiteralStringRef("bar")) == val);
+
+	state rocksdb::DB* db = ((RocksDBKeyValueStore*)kvStore)->db;
+	ASSERT(db != nullptr);
+
+	platform::eraseDirectoryRecursive("checkpoint");
+	state std::string checkpointDir = cwd + "checkpoint";
+
+	CheckpointRequest request(
+	    latestVersion, allKeys, RocksDBColumnFamily, deterministicRandom()->randomUniqueID(), checkpointDir);
+	CheckpointMetaData metaData = wait(kvStore->checkpoint(request));
+
+	state std::string rocksDBRestoreDir = "rocksdb-kvstore-br-restore-db";
+	platform::eraseDirectoryRecursive(rocksDBRestoreDir);
+
+	state IKeyValueStore* kvStoreCopy =
+	    new RocksDBKeyValueStore(rocksDBRestoreDir, deterministicRandom()->randomUniqueID());
+
+	wait(kvStoreCopy->restore(metaData));
+
+	Optional<Value> val = wait(kvStoreCopy->readValue(LiteralStringRef("foo")));
+	ASSERT(Optional<Value>(LiteralStringRef("bar")) == val);
+
+	return Void();
```

Comment:
Please erase any working directories so that subsequent unit tests can pass--there is a new TOML test file that executes all unit tests.

Reply:
Done.